### PR TITLE
Updating the custom accessibility object for DesignerTabButton to properly report 'Focusable'

### PR DESCRIPTION
**Pull request by tannergooding**
**Customer scenario**

Users who rely on the narrator will not know when the DesignerTabButtons get focus.

**Bugs this fixes:** 

https://devdiv.visualstudio.com/DevDiv/_workitems?id=409661

**Workarounds, if any**

Use a different narrator (such as NVDA)

**Risk**

Low. This just updates the accessibility object to properly report that the control is focusable.

**Performance impact**

None

**Is this a regression from a previous update?**

No

**Root cause analysis:**

It was thought that the Windows Narrator not reading the control text was due to a bug in Windows Narrator. It was later determined that it is because Windows Narrator is dependent on the accessibility object reporting that a control is focusable (where-as other narrators just read the text of the currently focused object).

**How was the bug found?**

Accessibility Tenet

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @dotnet/project-system, @dpoeschl

**Comment by tannergooding** [comment]
FYI. @MattGertz for accessibility approval

**Comment by srivatsn** [comment]
Can you please explain the fix?

**Comment by MattGertz** [comment]
(Tag me again when conversation resolves -- thx.)

**Comment by tannergooding** [comment]
@Srivatsn, the narrator checks the set of AccessibilityStates returned by the AccessibleObject to determine what the control supports.

We were only returning the Selected and Selectable states, so the Narrator would never check if the control got focus, even if it was given focus.

The fix updates the object to return the full set of accessibility states (this includes the 'Focusable' and 'Focused' states) so that the narrator does the right thing.

File: src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/DesignerTabButton.vb [hunk 1]
```diff
@@ -177,7 +177,7 @@ Namespace Microsoft.VisualStudio.Editors.ApplicationDesigner
 
         Friend ReadOnly Property DrawFocusCues As Boolean
             Get
-                Return ShowFocusCues OrElse (Focused And FocusedFromKeyboardNav)
+                Return ShowFocusCues OrElse (Focused AndAlso FocusedFromKeyboardNav)
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/DesignerTabButton.vb]: This is an internal property right - how is this related to the fix?

Comment (tannergooding) [inline src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/DesignerTabButton.vb]: This wasn't supposed to be part of the fix, will remove.

**Comment by tannergooding** [comment]
@Srivatsn, any other feedback here?
